kore/src/Kore/Step/Axiom/EvaluationStrategy.hs

```diff
@@ -256,13 +308,16 @@ applyFirstSimplifierThatWorks
                 orRemainders
             -- TODO (traiansf): this might generate too much output
             --    replace log with a logOnce when that becomes available
-            tryNextSimplifier
+            tryNextSimplifier Conditional
           | otherwise -> return applicationResult
-        AttemptedAxiom.NotApplicable -> tryNextSimplifier
+        AttemptedAxiom.NotApplicable -> tryNextSimplifier nonSimplifiability
```

Comment:
Just checking my understanding: If a rule doesn't match, that would be an example of a `NotApplicable` result, because the `SideCondition` doesn't determine matching, so changing it would never allow the match to succeed. On the other hand, if a rule matches, but the `requires` isn't satisfied, that would be an example of `NotApplicableUntilConditionChanges` because changing the `SideCondition` could allow the rule to apply. Does that sound right?

Reply:
Close. I added a comment.